packages/node_modules/@node-red/nodes/core/network/21-httpin.js

```diff
@@ -171,9 +171,18 @@ module.exports = function(RED) {
 
     if (RED.settings.httpNodeCors) {
         corsHandler = cors(RED.settings.httpNodeCors);
-        RED.httpNode.options("*",corsHandler);
     }
 
+    RED.httpNode.options("*", function(req,res,next) {
+        //see if any routes for this path exist & call next() otherwise call corsHandler
+        const routes = RED.httpNode._router.stack.filter(e =>  e.route && e.route.path == req.path && e.route.methods.options === true);
```

Comment:
I'm not sure this is going to quite work as intended.

The route path could be parameterised (`/foo/:id`) in which case you can't do a simple string compare with the request path. It also is quite an overhead to iterate over all the paths to find matches with every request.

A different approach would be to ensure the `corsHandler` is kept at the end of `RED.httpNode._router.stack` - so it acts as a catch-all in case no HTTP-In node gets there first. Its hacky as `_router.stack` is an internal of express... but I can't see how else to do this efficiently.

Reply:
AHH. Doh. It took some digging to get it here. There is the path regex but I tried to keep it light & missed the obvious (parametrised paths) 

With have a rethink on the whole approach. 

If you have any other ideas, please shout up.